api/tests/prescription/campaign/unit/application/api/campaigns-api_test.js

```diff
@@ -107,4 +108,47 @@ describe('Unit | API | Campaigns', function () {
       expect(result).not.to.be.instanceOf(Campaign);
     });
   });
+
+  describe('#getAll', function () {
+    it('should return paginated campaign list from organizationId', async function () {
+      const organizationId = Symbol('organizationId');
+      const page = Symbol('page');
+      const meta = Symbol('meta');
+      const campaignInformation1 = domainBuilder.buildCampaignReport({
+        id: 777,
+        code: 'SOMETHING',
+        name: 'Godzilla',
+        title: 'is Biohazard',
+        customLandingPageText: 'Rooooooooooooar',
+        createdAt: new Date('2020-01-01'),
+        archivedAt: new Date('2023-01-01'),
+      });
+      const campaignInformation2 = domainBuilder.buildCampaignReport({
+        id: 666,
+        code: 'SOMETHING',
+        name: 'Monarch',
+        title: 'is a butterfly',
+        customLandingPageText: 'pshh pshh pshhh pshhh',
+        createdAt: new Date('2020-01-01'),
+        archivedAt: new Date('2023-01-01'),
+      });
+
+      const getCampaignStub = sinon.stub(usecases, 'findPaginatedFilteredOrganizationCampaigns');
+      getCampaignStub
+        .withArgs({ organizationId, page })
+        .resolves({ models: [campaignInformation1, campaignInformation2], meta });
+
+      // when
+      const result = await campaignApi.getAll({ organizationId, page });
+
+      // then
+      const campaignList = result.models[0];
+      expect(result.meta).to.be.equal(meta);
+      expect(campaignList).not.to.be.instanceOf(CampaignReport);
```

Comment:
```suggestion
      expect(campaignList).to.be.instanceOf(CampaignListItem);
```

Ce ne serait pas plus clair en indiquant ce que c'est ?

Reply:
C'est qu'on veut explicitement montrer que ce n'est pas le model de notre Bounded Context. L'instance que l'on vous renvoi n'a pas d'importance.